### PR TITLE
fix(NotificationUtils): Encode user email to retrieve push devices.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [ ] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [ ] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Changes from this PR include correctly encoding a user's email address before retrieving the number of push devices.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java [hunk 1]
```diff
@@ -327,7 +329,7 @@ public static int getPushInfo(String toUser) {
         try {
             Map<String, String> headers = Map.of("Accept", "application/json");
             var httpResponse = HttpUtils.httpRequestRawResponse(
-                URI.create(PUSH_API_URL + "/devices/get?api_key=" + PUSH_API_KEY + "&user=" + toUser),
+                URI.create(getPushDevicesUrl(PUSH_API_URL + "/devices/get?api_key=" + PUSH_API_KEY + "&user=", toUser)),
```
Comment (br648) [inline src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java]: Nit. This might now benefit from a String.format?

**Review by br648 — APPROVED** [review]
LGTM

File: src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/utils/NotificationUtils.java]: Good idea, done in https://github.com/ibi-group/otp-middleware/pull/202/commits/0e8ba243917c8df7c997c4825efa222259f95153.

**Review by miles-grant-ibigroup — APPROVED** [review]
Can't test this but code looks good!

**Pull request merged**